ublox_serialization/include/ublox_serialization/serialization.hpp

```diff
@@ -387,6 +390,9 @@ class Reader {
           (count_ == 1 || data_[1] == options_.sync_b)) {
         break;
       }
+      else {
+        unused_data_.push_back(data_[0]);
```

Comment:
What happens if this data is not NMEA data, and further isn't even ASCII data?  What will trying to stuff it into a `std::string` do in that case?

Reply:
Yeah, I wondered about this while looking over the ROS 1 version as well. The UBX messages are a binary format rather than clear text, from what I understand, so any UBX messages not picked up by a callback will end up in the "unused" data. It then gets cast to a char and treated as whatever the ASCII equivalent of that byte is. The big risk then is if, when we're scrubbing through the buffer, one of those bytes happens to match a '$', because then we'd try to parse from there to the next newline as an NMEA string, and end up publishing nonsense.

I'm not sure exactly how to mitigate that, and haven't seen it occur in testing, but I'm certainly open to ideas.